tests/chart_tests/test_kube_state_deployment.py

```diff
@@ -83,3 +83,30 @@ def test_kube_state_deployment_singleNamespace(self, kube_version):
         )
         c_by_name = get_containers_by_name(docs[0])
         assert "--namespaces=test_namespace" in c_by_name["kube-state"]["args"]
+
+    def test_kube_state_default_collectors(self, kube_version):
+        docs = render_chart(
+            kube_version=kube_version,
+            values={},
+            show_only=["charts/kube-state/templates/kube-state-role.yaml"],
+        )
+
+        assert len(docs[0]["rules"]) == 23
+
+    def test_kube_state_specific_collector_enabled(self, kube_version):
+        docs = render_chart(
+            kube_version=kube_version,
+            values={"kube-state": {"collectors": ["certificatesigningrequests"]}},
+            show_only=["charts/kube-state/templates/kube-state-role.yaml"],
+        )
+
+        assert len(docs[0]["rules"]) == 1
```

Comment:
We can do better here since we know that `certificatesigningrequests` will be in the rules.

Reply:
```suggestion
        assert len(docs[0]["rules"]) == 1
        assert "certificatesigningrequests" in docs[0]["rules"][0]["resources"]
```